Update mueller.py for calibration matrix

WillisAllen opened the pull request:
Code implementation of the paper https://doi.org/10.1364/AO.46.008533 modified for Meuller matrices. Not sure if the number of singular values is correct, the paper uses 4 for calibrating a Stokes polarimeter, I used 9 or 16 depending if the matrix is (3,3) or (4,4). Also not sure how to save the W calibration matrix after it's calculated. This code is untested.

WillisAllen committed:
Update mueller.py for calibration matrix

WillisAllen committed:
Fix formatting and # of singular values to keep in mueller.py

WillisAllen committed:
Spelling error mueller.py

WillisAllen committed:
Update mueller.py

Add methods to save and load calibration W matrix.

WillisAllen committed:
Update mueller.py

Switch order of PSG and PSA in calibration method parameters

WillisAllen committed:
Update mueller.py

WillisAllen committed:
Update README.md

Add sections to explain how to make a calibration matrix and how to use one.

WillisAllen committed:
Update README.md

Formatting

WillisAllen committed:
Update README.md

Crayfi commented:
I've added example code and an explanation for how to produce a calibration matrix and save it to a file then load and use it to calculate the Mueller matrix. Also fixed some errors in the formatting and added in the methods to save and load a calibration file. Not sure how to save json files so I implemented it using a csv.

WillisAllen committed:
Update README.md

WillisAllen committed:
Update mueller.py

Corrected linear algebra equations to reflect the paper using a single reference Mueller matrix instead of the two used for the original calcMueller(). The equation on line 36 is the correct equation but I would greatly appreciate if somebody could verify the code actually implements this equation.

WillisAllen committed:
Update mueller.py

Changed the equations to use N and M for number of measurement configurations and number of reference samples instead of the specific numbers I'm using.

WillisAllen committed:
Update README.md

Changed from two reference matrices to the single matrix to reflect how the research paper calculates W.

WillisAllen committed:
Added details about calibrating with reference samples

WillisAllen committed:
Update README.md

Formatting edit

WillisAllen committed:
Update mueller.py

Forgot a couple of lines from the original code

WillisAllen committed:
Update README.md note about multipsectral instruments

WillisAllen committed:
Update mueller.py remove calibration funcitons

Moving calibration functions to their own file.

WillisAllen committed:
Create calibration

Moved all the calibration functions here and added in a function to calculate the Stokes vector using a calibration matrix as well.

WillisAllen committed:
Rename calibration to calibration.py

WillisAllen committed:
Update README.md Add Stokes vector calibration 

Added details for how to make a Stokes vector calibration.

WillisAllen committed:
Update calibration.py

Typos

WillisAllen committed:
Update calibration.py

Add number to use for k in svdpinv for Stokes, 4, from research paper

WillisAllen committed:
Update calibration.py

Formatting

WillisAllen committed:
Update calibration.py

More formatting

Crayfi commented:
The calibration file I added should allow for calibration of Mueller matrices and Stokes vectors with the use of reference samples. Code is untested.
